--- origsrc/libmtp-0.3.5/configure.ac	2008-12-13 17:56:06.000000000 -0600
+++ src/libmtp-0.3.5/configure.ac	2008-12-28 21:47:04.643750000 -0600
@@ -61,7 +61,8 @@
 # zlib.h the day we need to decompress firmware
 AC_CHECK_HEADERS([ctype.h errno.h fcntl.h getopt.h libgen.h \
 	limits.h stdio.h string.h sys/stat.h sys/time.h unistd.h \
-	iconv.h langinfo.h locale.h])
+	langinfo.h locale.h])
+AM_ICONV
 AC_CHECK_HEADER([usb.h],,
 	AC_MSG_ERROR([I can't find the libusb header file on your system.
 	You may need to set the CPPFLAGS environment variable to include
--- origsrc/libmtp-0.3.5/src/Makefile.am	2008-12-21 07:07:06.000000000 -0600
+++ src/libmtp-0.3.5/src/Makefile.am	2008-12-28 21:48:12.628125000 -0600
@@ -51,4 +51,5 @@
 endif
 
 libmtp_la_LDFLAGS=@LDFLAGS@  -no-undefined -export-symbols $(srcdir)/libmtp.sym -version-info $(W32LF) $(SOVERSION)
+libmtp_la_LIBADD = $(LTLIBICONV)
 libmtp_la_DEPENDENCIES=libmtp.sym
