--- origsrc/libmtp-0.3.0/configure.ac	2008-06-08 16:37:36.000000000 -0500
+++ src/libmtp-0.3.0/configure.ac	2008-07-05 23:33:54.968750000 -0500
@@ -61,7 +61,8 @@
 # zlib.h the day we need to decompress firmware
 AC_CHECK_HEADERS([ctype.h errno.h fcntl.h getopt.h libgen.h \
 	limits.h stdio.h string.h sys/stat.h sys/time.h unistd.h \
-	iconv.h langinfo.h])
+	langinfo.h])
+AM_ICONV
 AC_CHECK_HEADER([usb.h],,
 	AC_MSG_ERROR([I can't find the libusb header file on your system.
 	You may need to set the CPPFLAGS environment variable to include
--- origsrc/libmtp-0.3.0/libmtp.pc.in	2007-11-10 18:06:28.000000000 -0600
+++ src/libmtp-0.3.0/libmtp.pc.in	2008-07-05 23:34:10.265625000 -0500
@@ -8,8 +8,6 @@
 Name: libmtp
 Description: libmtp is a library for accessing Media Transfer Protocol devices
 Version: @VERSION@
-Requires: libusb
-Conflicts:
 Libs: -L${libdir} -lmtp
 Libs.private @LIBS@
 Cflags: -I${includedir} @OSFLAGS@
--- origsrc/libmtp-0.3.0/src/Makefile.am	2008-06-23 17:56:34.000000000 -0500
+++ src/libmtp-0.3.0/src/Makefile.am	2008-07-05 23:34:10.265625000 -0500
@@ -35,7 +35,7 @@
 
 if COMPILE_MINGW32
 noinst_DATA=libmtp.lib
-W32LF=-export-dynamic -no-undefined -export-symbols libmtp.sym
+W32LF=-export-symbols libmtp.sym
 libmtp.def: libmtp.sym
 	echo "LIBRARY \"@PACKAGE@\"" > libmtp.def
 	echo "DESCRIPTION \"Media Transfer Protocol (MTP) library\"" >> libmtp.def
@@ -50,4 +50,5 @@
 	$(INSTALL) libmtp.lib $(DESTDIR)$(libdir)
 endif
 
-libmtp_la_LDFLAGS=@LDFLAGS@ -version-info $(SOVERSION) $(W32LF)
+libmtp_la_LDFLAGS=@LDFLAGS@ -version-info $(SOVERSION) -export-dynamic -no-undefined $(W32LF)
+libmtp_la_LIBADD=$(LTLIBICONV)
--- origsrc/libmtp-0.3.0/src/ptp.c	2008-05-28 15:26:24.000000000 -0500
+++ src/libmtp-0.3.0/src/ptp.c	2008-07-05 23:34:10.281250000 -0500
@@ -1176,7 +1176,7 @@
  *
  */
 uint16_t
-ptp_ek_setserial (PTPParams* params, unsigned char *data, unsigned int size)
+ptp_ek_setserial (PTPParams* params, unsigned char *data, uint32_t size)
 {
 	PTPContainer ptp;
 
